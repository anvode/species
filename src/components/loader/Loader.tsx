import * as React from 'react';
import { useContext } from 'react';

import { SpeciesContext } from '../../context/species/SpeciesContext';

import './Loader.scss';

const Loader = () => {
    const { state: {speciesLoadingText} } = useContext(SpeciesContext);

    return (
        <div className="loader__container">
            <h2 className="loader__title">{speciesLoadingText}</h2>
            <div className="loader"><div></div><div></div><div></div><div></div></div>
        </div>
    );
};

export default Loader;
