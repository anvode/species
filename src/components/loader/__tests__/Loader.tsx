import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import Loader from '../Loader';

afterEach(cleanup);

describe('Loader', () => {
    it('snapshot Loader', () => {
        const { asFragment } = render(<Loader />);
        expect(asFragment()).toMatchSnapshot();
    });
});

