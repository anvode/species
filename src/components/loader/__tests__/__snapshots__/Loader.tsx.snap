// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader snapshot Loader 1`] = `
<DocumentFragment>
  <div
    class="loader__container"
  >
    <h2
      class="loader__title"
    />
    <div
      class="loader"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
